interceptor.go

```diff
@@ -25,12 +25,13 @@ type interceptor struct {
 
 // Interceptor is something that can intercept HTTP requests.
 type Interceptor interface {
-	Intercept(w http.ResponseWriter, req *http.Request, forwardInitialRequest bool, op ops.Op, defaultPort int)
+	Pipe(op ops.Op, w http.ResponseWriter, req *http.Request, defaultPort int, dial func(network, addr string) (net.Conn, error))
```

Comment:
Now the two methods share very few common code and do totally different things. Maybe split to different interfaces with more proper name?